components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/validator/impl/SigningAlgorithmValidator.java

```diff
@@ -0,0 +1,93 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.identity.extensions.validator.impl;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.apache.commons.beanutils.NestedNullException;
+import org.apache.commons.beanutils.PropertyUtilsBean;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.bfsi.consent.management.common.exceptions.ConsentManagementException;
+import org.wso2.bfsi.consent.management.common.util.CommonUtils;
+import org.wso2.bfsi.identity.extensions.util.IdentityCommonConstants;
+import org.wso2.bfsi.identity.extensions.util.IdentityCommonUtils;
+import org.wso2.bfsi.identity.extensions.validator.annotation.ValidSigningAlgorithm;
+import org.wso2.carbon.identity.oauth2.RequestObjectException;
+
+import java.lang.reflect.InvocationTargetException;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * To validate if the signing algorithm used to sign request object
+ * is the same as the algorithm given during registration.
+ */
+public class SigningAlgorithmValidator implements ConstraintValidator<ValidSigningAlgorithm, Object> {
+
+    private String algorithmXpath;
+    private String clientIdXPath;
+    private static Log log = LogFactory.getLog(SigningAlgorithmValidator.class);
+
+    @Override
+    public void initialize(ValidSigningAlgorithm constraintAnnotation) {
+
+        this.algorithmXpath = constraintAnnotation.algorithm();
+        this.clientIdXPath = constraintAnnotation.clientId();
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+
+        try {
+            final String algorithm = new PropertyUtilsBean().getProperty(object, algorithmXpath).toString();
+            final String clientId = BeanUtils.getProperty(object, clientIdXPath);
+
+            return algorithmValidate(algorithm, clientId);
+
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException | NestedNullException e) {
+            log.error("Error while resolving validation fields", e);
+            return false;
+        }
+    }
+
+     boolean algorithmValidate(String requestedAlgo, String clientId) {
+
+        try {
+            if (!(StringUtils.isNotEmpty(IdentityCommonUtils.getCertificateContent(clientId))
+                    && CommonUtils.isRegulatoryApp(clientId))) {
+                String registeredAlgo = IdentityCommonUtils.getAppPropertyFromSPMetaData(
+                        clientId, IdentityCommonConstants.REQUEST_OBJECT_SIGNING_ALG);
+
+                if (StringUtils.isBlank(registeredAlgo)) {
+                    // TODO: check with DCR API
```

Comment:
Let's create an issue and add the link here if required.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136